app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -345,6 +351,80 @@ protected function getCorrectedTotal($projectedTotal, $magentoDerivedCartData)
         return $magentoDerivedCartData;
     }
 
+    /**
+     * Checks if billing address of a quote has all the expected fields.
+     * If not, the address is updated by using the provided shipping address.
+     *
+     * @param Mage_Sales_Model_Quote $quote Quote that is expected to contain both
+     *                                      a billing and shipping address.
+     *
+     * @return bool  true if a correction is made, otherwise false
+     */
+    public function correctBillingAddress($quote)
+    {
+        $billingAddress = $quote->getBillingAddress();
+        $shippingAddress = $quote->getShippingAddress();
+        $wasCorrected = false;
+
+        /////////////////////////////////////////////
+        /// Changes persisted to Magento and Bolt
```

Comment:
why some change is saved to magento while other is not?

Reply:
I agree. If there's a specific reason it should be onlined why in a comment.

Reply:
ok.  I'll add more comments.